### PR TITLE
330 fix/delete user meta add additional tests

### LINKED ISSUE #330: Delete user meta - Add additional tests
Add these additional tests to the Delete User Meta module.

- [x] Test deletion of user metas from more than one users in a single role.
- [x] Test deletion of user metas from more than role, with easy role having one user
- [x] Test deletion of user metas from more than role, with easy role having more than one user
- [x] Test deletion of user metas from one role, which has no users. Nothing should be deleted.
- [x] Test deletion of user metas from more than one role, where each role has no users. Nothing should be deleted.
- [x] Test deletion of user metas from more than one role, where one role doesn't have any users and other role has users. Nothing should be deleted.
- [x] Test deletion of user metas with both meta key and value (might require add-on for this)

### PR DESCRIPTION
**Pull request by rajanit2000**
Fix #330

### CODE REVIEW

**Comment by rajanit2000** [comment]
@sudar development is done, Its ready to review

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 1]
```diff
@@ -20,6 +20,11 @@ class DeleteUserMetaModuleTest extends WPCoreUnitTestCase {
 	 */
 	protected $module;
 
+	/**
+	 * Setup method
+	 *
+	 * @return void
```
Comment (sudar) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Don't add `@return` if there is nothing to return.

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 2]
```diff
@@ -153,4 +158,224 @@ public function test_delete_usermeta_in_batches() {
 		$meta_deleted = $this->module->delete( $delete_options );
 		$this->assertEquals( 10, $meta_deleted );
 	}
+
+	/**
+	 * Test deletion of user metas from more than one users in a single role.
+	 */
+	public function test_deleting_multiple_users_with_meta_fields_from_single_user_role() {
```
Comment (sudar) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Change the test method format to `test_that_` format.

I know that other methods have old format, but do it for all the new methods that you have created.

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 3]
```diff
@@ -153,4 +158,224 @@ public function test_delete_usermeta_in_batches() {
...
+
+		// Create a users with meta value in admin role.
+		for ( $i = 0; $i < 10; $i++ ) {
```
Comment (sudar) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Always add atleast one more user who will not be deleted.

**Comment by sudar** [comment]
@rajanit2000 

Like I told you yesterday, always add at least one more data that will not be deleted to the test case.

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 4]
```diff
@@ -29,18 +32,20 @@ public function setUp() {
 	/**
 	 * Add to test single user meta from admin user role.
 	 */
-	public function test_deleting_single_user_meta_fields_from_admin_user_role() {
+	public function test_that_delete_single_user_meta_fields_from_admin_user_role() {
```
Comment (sudar) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: This naming is wrong.

It should be `test_that_meta_fields_from_a_single_user_in_admin_role_can_be_delete`

Change it for other methods.

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 5]
```diff
@@ -152,5 +182,294 @@ public function test_delete_usermeta_in_batches() {
...
+	 *
+	 * @param int $user_id User id.
+	 */
+	public function is_user_id_exists( $user_id ) {
+
+		global $wpdb;
+
+		$count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM $wpdb->users WHERE ID = %d", $user_id ) );
+
+		if ( 1 == $count ) {
```
Comment (sudar) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Replace it with just `return ($count > 0)`

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 6]
```diff
@@ -49,23 +54,30 @@ public function test_deleting_single_user_meta_fields_from_admin_user_role() {
 
 		$user_meta = get_user_meta( $user, 'time', true );
 		$this->assertEquals( '', $user_meta );
+
```
Comment (sudar) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: We are deleting user meta and not users. So you should add another user meta for the same user and make sure that the additional meta is not deleted.

Do this for all the test cases.

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php (hunk 4 above)
Comment (rajanit2000) [reply tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Fixed

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php (hunk 5 above)
Comment (rajanit2000) [reply tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Fixed

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php (hunk 6 above)
Comment (rajanit2000) [reply tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Fixed

**Comment by rajanit2000** [comment]
@sudar fix the review changes, Ready to Review

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 7]
```diff
@@ -29,18 +32,20 @@ public function setUp() {
...
 	 * Add to test single user meta from admin user role.
 	 */
-	public function test_deleting_single_user_meta_fields_from_admin_user_role() {
+	public function test_that_meta_fields_from_a_single_user_in_admin_role_can_be_delete() {
 		// Create a user with admin role.
 		$user = $this->factory->user->create( array( 'role' => 'administrator' ) );
 
 		add_user_meta( $user, 'time', '10/10/2018' );
 
+		add_user_meta( $user, 'another', 'Another' );
```
Comment (sudar) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: In `add_user_meta` you are using the meta value `Another`. But in `delete_user_meta` you are using the value `another`.

Store the meta value in a variable and then try to use the variable everywhere.

This is happening in other test cases as well. Fix in all of them.

Comment (rajanit2000) [reply tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Fixed

**Comment by rajanit2000** [comment]
@sudar its ready to review

File: tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php [hunk 8]
```diff
@@ -152,5 +191,301 @@ public function test_delete_usermeta_in_batches() {
...
+			'delete_options' => '',
+		);
+
+		$meta_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 1, $meta_deleted );
+
+		$user_meta = get_user_meta( $user, 'test_key', true );
+		$this->assertEquals( '', $user_meta );
+
+		$exist_user = $this->is_user_meta_exists( $user, $another_meta_key );
```
Comment (rajanit2000) [inline tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php]: Need to update by value

**Comment by rajanit2000** [comment]
@sudar i have fix the review changes, And also fix the delete user meta by using value issue

**Pull request merged**